src/get_mounts/bsd.rs

```diff
@@ -15,24 +14,21 @@ extern "C" {
     fn getmntinfo(mntbufp: *mut *mut statfs, flags: c_int) -> c_int;
 }
 
-pub fn get_mount_points() -> Vec<String> {
+pub fn get_mount_points() -> Vec<PathBuf> {
     // FIXME: move this into a Darwin-specific module & implement the Linux version
     let mut raw_mounts_ptr: *mut statfs = ptr::null_mut();
 
     let rc = unsafe { getmntinfo(&mut raw_mounts_ptr, MNT_NOWAIT) };
-
-    if rc < 0 {
-        panic!("getmntinfo returned {:?}", rc);
-    }
+    assert!(rc >= 0, "getmntinfo returned {:?}");
```

Comment:
This matches the existing implementation, but the [man page](https://www.freebsd.org/cgi/man.cgi?query=getmntinfo#RETURN_VALUES) seems to indicate a 0 value is an error as well. Perhaps the get_mount_points() functions should return an io::Result. If 0 is returned, we could use `io::Error::last_os_error` to get errno

Reply:
You're right — probably unsurprising given that I haven't found a non synthetic way to trigger a failure in that API but it should definitely check `errno`.